docs/dev/framework/security.md

```diff
@@ -0,0 +1,275 @@
+
+    encoders:
+        Contao\User:
+            algorithm: auto
+
+    firewalls:
+        dev:
+            pattern: ^/(_(profiler|wdt|error)|css|images|js)/
+            security: false
+
+        contao_install:
+            pattern: ^/contao/install$
+            security: false
+
+        contao_backend:
+            request_matcher: contao.routing.backend_matcher
+            provider: contao.security.backend_user_provider
+            user_checker: contao.security.user_checker
+            anonymous: ~
+            switch_user: true
+
+            contao_login:
+                remember_me: false
+
+            logout:
+                path: contao_backend_logout
+                handlers:
+                    - contao.security.logout_handler
+                    - contao_manager.security.logout_handler
+                success_handler: contao.security.logout_success_handler
+
+        contao_frontend:
+            request_matcher: contao.routing.frontend_matcher
+            provider: contao.security.frontend_user_provider
+            user_checker: contao.security.user_checker
+            anonymous: ~
+            switch_user: false
+
+            contao_login:
+                remember_me: true
+
+            remember_me:
+                secret: '%kernel.secret%'
+                remember_me_parameter: autologin
+
+            logout:
+                path: contao_frontend_logout
+                target: contao_root
+                handlers:
+                    - contao.security.logout_handler
+                success_handler: contao.security.logout_success_handler
+
+    access_control:
+        - { path: ^/contao/login$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^/contao/logout$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
+        - { path: ^/contao(/|$), roles: ROLE_USER }
+        - { path: ^/, roles: [IS_AUTHENTICATED_ANONYMOUSLY] }
+```
+{{% /expand %}}
+
+If you want to learn more about Symfony's Security Component use the provided links to read up on. This documentation will only cover
+implmentation details that are unique to Contao.
+
+Since within Contao you can put a login form on basically any page, Contao does not utilise Symfony's built-in 
+[`form_login` Authentication Provider][SymfonyFormLogin]. Instead, Contao implements its own user providers, 
+[user checkers][SymfonyUserChecker] and [request matcher service][SymfonyRequestMatcherService], the latter of which checks the 
+[request scope][RequestScope]. For example in the front end, all URLs to Contao pages will have the `_scope` request attribute set to 
+`frontend` and the `contao_frontend` firewall will thus be applicable to all these URLs.
+
+
+## Voters
+
+Starting with Contao **4.7** Contao implements so called [Voters][SymfonyVoters] in order to easily check whether an authenticated user is
+authorized to access specific resources. These voters are automatically used by Symfony via the [Security Helper][SecurityHelperService].
```

Comment:
Not sure what your definition of "automatically used" is. If noone calls `isGranted` then they are obviously not used.

Reply:
Meaning that no body calls your voter directly, it will be automatically called when using `isGranted` of the security helper, if the given attribute is handled by your voter.

Reply:
ok… I would find that rather confusing. _Obviously_ a voter is supposed to be used if you register one, the term _automatically_ implies (to me) that I don't have to check permissions if I implement a voter. Maybe something like _You can use the Contao voters in your code to check permissions._

Reply:
But you never use a voter yourself? I would find the latter confusing. You implement a voter and then Symfony tags the voter's service and injects it elsewhere, so that they are invoked whenever `isGranted` is called.

Reply:
that's true, it would rather be _use Symfony security to check permissions_ then?

Reply:
I cannot find a way to write a sentence without _automatically_ 😁. For me, Symfony automatically uses these Voters for me. All I do is create the class and nothing else (with autowiring and autoconfiguration enabled). I don't have to tell Symfony anything to use the voter, Symfony uses them _automatically_ when checking for permissions.

Reply:
Symfony _will_ add the voter to the security system "automatically". But it will not check for any permission related to the new voter, if noone will ask for it. And asking for a permission your voter added is always your job.